### ISSUE TITLE
Meower seems to be unsafe, regarding consistency

### ISSUE THREAD

**Issue by arnaud-lb** [tinrab/meower#1]
It seems that the query tier can get inconsistent quite easily: Some meows may not be available through the query tier.

For instance, [here](https://github.com/tinrab/meower/blob/157e22b6b0d4490d7aa5120818706cf98735a612/meow-service/handlers.go#L49) an event is published through NATS (I believe), but if something goes wrong, the query tier will never hear of this meow.

Later, in the query tier, an attempt is made to index a meow in Elasticsearch. If it [fails](https://github.com/tinrab/meower/blob/157e22b6b0d4490d7aa5120818706cf98735a612/query-service/handlers.go#L24), the neow will never be available in search.

**Comment by tinrab** [comment]
Yes, all this can happen. It could be resolved with a [saga](http://microservices.io/patterns/data/saga.html). I have a possible implementation on my To-Do list :smile_cat:

**Comment by gedw99** [comment]
Would event store pattern help ?

Example:
https://medium.com/@ivanjaros/diy-startup-simple-event-store-with-boltdb-e8146d386753

**Comment by tinrab** [comment]
@gedw99 Yes, if you can guarantee that events will be delivered at least once. If the event gets properly handled on the query side, then the delivery can be considered successful.

**Comment by gedw99** [comment]
Recently i started experimenting with the whole issue of co-ordination or data ( and single source of it) using CRDT and automerge. Its still not 100% solved architecturally though.
